tests/test_bad_dags.py

```diff
@@ -36,8 +37,10 @@
 def param_id(param):
     if inspect.isclass(param):
         return param.__name__
-    else:
+    elif isinstance(param, six.string_types):
```

Comment:
Do we need whole six package to determine that this value is a string?  Just use `isinstance(param, type(u'')` to catch that case.

Reply:
`six` is already installed in tests as a transitive dependency.

`type(u'')` is definitely an option, but it would be easier to miss it when python 2.7 support will be dropped. `six` OTOH, is easier to grep for.

Reply:
2.7 support may be dropped with Airflow 2.0, but if they decide to take that blame for just one single year, so it will happens only with Airflow 3.0 release.

six may be transitive dependency, but with this move it becames first party one just for a single case. While yes, it's easy to forget about come "strage code" to cleanup additional dependency requires more maintenance work to do. Imagine conflict on six version between first and third parties. It's a well known case when six break down the API causing incompatibility issues.

Reply:
> it becames first party one just for a single case

There might become more of them as the project would move forward.

> additional dependency requires more maintenance work to do.

That is true in general, but for this specific case I think it's on the contrary: `six`, I believe, would ease maintenance (and further move from py2+3 to py3 only) by being a well-known and clear marker of py2-3 compatibility junction points (as opposed to the `type(u'')` trickery).

> Imagine conflict on six version between first and third parties.

This is quite unlikely, given that `six` haven't had new releases for a long time. Even if there would be such a conflict, it would appear only in the develop mode, which won't affect anyone but the maintainers of the project.

> It's a well known case when six break down the API causing incompatibility issues.

Interesting, I never heard about this. Could you please provide an example?

Reply:
Or we can drop 2.7 support right away, assuming that no-one has actually been using this project before. This would be way easier for maintenance than everything being discussed in this thread.

Reply:
I wouldn't care much about 2.7, but take a whole dependency just for that case is a big overhead. Actually, we can drop it with easy once we remove all the 2.x bits which uses declarative dags in our production. If someone wanted to keep it support - patches welcome, but no warantees.

However, there shouldn't be much cases for Python 2 vs Python 3 differences, so those could be simply avoided by tricks like `type(u'')` - those are cheap.

Reply:
> I wouldn't care much about 2.7, but take a whole dependency just for that case is a big overhead.

I don't quite agree with this statement, given that `six` is already getting installed, so nothing would actually be changed.

But to avoid disputes over insignificant stuff I'd accept your position: I've just pushed the `type(u'')` option.

Reply:
Thank you. To be clear, I don't want to make codebase knows too much about 2vs3 differences. Six forces to embed them. So far we have only one precedence and just becase of tests. Let's keep going to see how much they'll be. It wouldn't be hard to go back to six or more modern  futures if we face something too specific (like  `raise from`) to reimplement on our own.